### DISCUSSION TITLE
[Solved] How to package the electron app embedding ffmpeg?

### DISCUSSION THREAD

**Discussion by junoyoon** [padvincenzo/silence-speedup#6] [Q&A] [answered]
@padvincenzo

First of all, I'd like to say THANKS for this great project.
I'm really happy to use this tool to shorten my video. I saved my effort a lot and I hope to spread this to my colleagues who are suffering video editing hell.

One thing I hope to see is.. mac silicon native support version (m1)
I was succeed run this project by ```npm start``` in my m1 machine and learned how to use electron packagers.
However, One thing I missed is how to embed ffmpeg into final binary file.

Could you explain how to do it?

**Comment by padvincenzo** [comment, marked as answer]
Hi @junoyoon
Thanks for your feedback

To embedd FFmpeg (a static build of it), before executing electron-packager, you have to move the executable (and preferable its license file) in the directory `assets/ffmpeg`. This is the way I embed FFmpeg in the standalone versions.

Silence SpeedUp will then detect the presence of that executable, and will use it, unless you have changed the preferences. To be sure, before using electron-packager, open the file `config.json` and delete the value of `ffmpegPath` (and `exportPath` if you wish) (I mean, make them empty strings).

Hope this help

**Reply by junoyoon** [reply]
Thank you very much.

**Reply by junoyoon** [reply]
@padvincenzo 

Which file should locate there?

I found libffmpeg.dylib is in the app file.

![image](https://user-images.githubusercontent.com/1000705/150085719-25d837e5-b0a2-487f-baf6-5dd4e5b38c0f.png)

Is it enough to locating libffmpeg.dylib in ```assets/ffmpeg```?

**Comment by padvincenzo** [comment]
Hi @junoyoon 
Not at all, you have to [download FFmpeg from its website](https://ffmpeg.org/download.html), either a static build or the source code.

If the static build works then just copy the executable of ffmpeg in the `assets/ffmpeg` directory of Silence SpeedUp.

If the static build doesn't work, you have to download the source code of ffmpeg and compile and install on your pc. I've found a video tutorial on youtube that may help: https://www.youtube.com/watch?v=v_tssMtCP0s . When ffmpeg is installed, found the executable in `/usr/local/bin/` and copy it in the `assets/ffmpeg` directory of Silence SpeedUp.

**Comment by junoyoon** [comment]
Thx for explanation.  
Following your guideline, I'm succeed to packaging it with m1's ffmpeg binary.
I don't know how to build ffmpeg.. So I just installed ffmpeg by brew and copied the installed ffmpeg executable to ```asserts/ffmpeg```

And I did some experiment. 
it tooks **1 min** for processing 7m30s video with m1 native. It was **1m40s** with intel version.
Thx!!!!

**Comment by padvincenzo** [comment]
Happy to hear that, I'm marking this as solved.
